platform/wayland/cog-platform-wl.c

```diff
@@ -1460,10 +1455,8 @@ static const struct wp_presentation_feedback_listener presentation_feedback_list
 static void
 request_frame(void)
 {
-    if (wpe_view_data.frame_callback == NULL) {
-        wpe_view_data.frame_callback = wl_surface_frame(win_data.wl_surface);
-        wl_callback_add_listener(wpe_view_data.frame_callback, &frame_listener, NULL);
-    }
+    struct wl_callback *frame_callback = wl_surface_frame(win_data.wl_surface);
+    wl_callback_add_listener(frame_callback, &frame_listener, NULL);
```

Comment:
Removing the variable and the null-check can result in more than one frame callback being registered for the same frame. The check was there to allow calling `request_frame()` from different places but still having only one callback registered for each frame.

Are you sure that without the check we never schedule more than one callback per frame?